store/src/java/com/zimbra/cs/service/mail/CreateDataSource.java

```diff
@@ -252,8 +254,23 @@ static void validateFolderId(Account account, Mailbox mbox, Element eDataSource,
             }
         }
     }
+    /**
+     * Confirms that the zimbraDataSourceEmailAddress attribute is unique
+     * 
+     */
+    public static void validateDataSourceEmail(Account account, Element eDataSource)
```

Comment:
As this `validateDataSourceEmail(Account account, Element eDataSource)` method is used only for `ModifyDataSource`Class and indirectly called from `processCommonOptionalAttrs()` so its better you can define this method in that class as well. unnecessarily you are creating this in `CreateDataSource` class and calling it from `ModifyDataSource` class

Reply:
better to move to utility class.

Reply:
fixed